### PR TITLE
DOC: Add details how to write metadata

### PR DESCRIPTION
**Pull request by Ineffable22**
This commit introduces significant enhancements to the existing code for adding metadata to a PDF file. The changes aim to improve the readability, completeness, and professionalism of the metadata properties. The following updates have been made:
- Added new metadata properties, including Title, Subject, Keywords, CreationDate, ModDate, Creator, and CustomField, providing more comprehensive information about the PDF document.
- Refactored the code to utilize the datetime module to obtain the current date and time in the desired format, ensuring accurate and standardized timestamping.
- Introduced an optional feature to include UTC time, enabling precise timestamping in different time zones if required.
- Preserved the ability to retain old metadata by including the metadata variable, allowing seamless integration with existing PDF files.
- Ensured compatibility with the latest version of the pypdf library, guaranteeing optimal performance and support for future updates.

These modifications significantly enhance the functionality and professionalism of the code for manipulating PDF files. Thorough testing has been conducted to ensure the accuracy and reliability of the changes.

### CODE REVIEW

File: docs/user/metadata.md [hunk 1]
```diff
@@ -31,11 +32,26 @@ writer = PdfWriter()
...
     writer.add_page(page)
 
-# Add the metadata
+# If you want to add the old metadata, include this line
+metadata = reader.metadata
+writer.add_metadata(metadata)
+
+# Format the current date and time for the metadata
+utc_time = '\05505\04700\047' # UTC time optional
+time = datetime.now().strftime(f'D\072%Y%m%d%H%M%S{utc_time}')
```
Comment (MartinThoma) [inline docs/user/metadata.md:41]: What is the advantage of writing it like this compared to the simpler way:

```suggestion
utc_time = "-05'00'" # UTC time optional
time = datetime.now().strftime(f'D\072%Y%m%d%H%M%S{utc_time}')
```

File: docs/user/metadata.md [hunk 2]
```diff
@@ -31,11 +32,26 @@ writer = PdfWriter()
 for page in reader.pages:
     writer.add_page(page)
 
-# Add the metadata
+# If you want to add the old metadata, include this line
+metadata = reader.metadata
+writer.add_metadata(metadata)
+
+# Format the current date and time for the metadata
+utc_time = '\05505\04700\047' # UTC time optional
```
Comment (MartinThoma) [inline docs/user/metadata.md:40]: What is the advantage of writing it like this compared to the simpler way:

```suggestion
utc_time = "-05'00'" # UTC time optional
```

**Comment by MartinThoma** [comment]
Thank you for your contribution :pray: 

If you want, I can add you to https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html :-)

**Comment by Ineffable22** [comment]
That's fantastic! I would be honored to be added to the list of contributors on https://pypdf.readthedocs.io/en/latest/meta/CONTRIBUTORS.html. Thank you for the opportunity and recognition. It means a lot to me.

**Comment by MartinThoma** [comment]
I've just added you - it might take ~10 minutes until the docs are rebuilt :-)

**Pull request merged**